datamap/example/src/main/java/org/apache/carbondata/datamap/minmax/AbstractMinMaxDataMapWriter.java

```diff
@@ -0,0 +1,248 @@
+      if (DataTypes.BOOLEAN == dataType) {
+        indexColumnMinMaxCollectors[indexColIdx].update(
+            BooleanConvert.boolean2Byte((boolean) value));
+      } else if (DataTypes.SHORT == dataType) {
+        indexColumnMinMaxCollectors[indexColIdx].update((short) value);
+      } else if (DataTypes.INT == dataType) {
+        indexColumnMinMaxCollectors[indexColIdx].update((int) value);
+      } else if (DataTypes.LONG == dataType || DataTypes.TIMESTAMP == dataType) {
+        indexColumnMinMaxCollectors[indexColIdx].update((long) value);
+      } else if (DataTypes.DOUBLE == dataType) {
+        indexColumnMinMaxCollectors[indexColIdx].update((double) value);
+      } else if (DataTypes.isDecimal(dataType)) {
+        indexColumnMinMaxCollectors[indexColIdx].update((BigDecimal) value);
+      } else {
+        throw new UnsupportedOperationException("unsupported data type " + dataType);
+      }
+    } else {
+      // While pruning for query, we want to reuse the pruning method from carbon, so here for
+      // dictionary columns, we need to store the mdk value in the minmax index.
+      // For direct generating, the input value is already MDK; For late building, the input value
+      // is surrogate key, so we need to handle it here.
+      if (indexCol.hasEncoding(Encoding.DICTIONARY)
+          || indexCol.hasEncoding(Encoding.DIRECT_DICTIONARY)) {
+        indexColumnMinMaxCollectors[indexColIdx].update(convertDictValueToMdk(indexColIdx, value));
+      } else {
+        byte[] plainValue = convertNonDicValueToPlain(indexColIdx, (byte[]) value);
+        indexColumnMinMaxCollectors[indexColIdx].update(plainValue);
+      }
+    }
+  }
+
+  protected abstract byte[] convertDictValueToMdk(int indexColIdx, Object value);
+
+  protected abstract byte[] convertNonDicValueToPlain(int indexColIdx, byte[] value);
+
+  private void logMinMaxInfo(int indexColId) {
+    CarbonColumn indexCol = indexColumns.get(indexColId);
+    StringBuilder sb = new StringBuilder("flush blockletId->").append(currentBlockletId)
+        .append(", column->").append(indexCol.getColName())
+        .append(", dataType->").append(indexCol.getDataType().getName());
+    Object min = indexColumnMinMaxCollectors[indexColId].getPageStats().getMin();
+    Object max = indexColumnMinMaxCollectors[indexColId].getPageStats().getMax();
+    if (indexCol.isMeasure()
+        || (indexCol.isDimension()
+        && DataTypeUtil.isPrimitiveColumn(indexCol.getDataType())
+        && !indexCol.hasEncoding(Encoding.DICTIONARY)
+        && !indexCol.hasEncoding(Encoding.DIRECT_DICTIONARY))) {
+      sb.append(", min->").append(min)
+          .append(", max->").append(max);
+    } else {
+      sb.append(", min->")
+          .append(new String((byte[]) min, CarbonCommonConstants.DEFAULT_CHARSET_CLASS))
+          .append(", max->")
+          .append(new String((byte[]) max, CarbonCommonConstants.DEFAULT_CHARSET_CLASS));
+
+    }
+    LOGGER.debug(sb.toString());
```

Comment:
Please add a check for isDebugEnabled() wherever debug logs added

Reply:
yeah, actually this whole method `logMinMaxInfo` is already guarded by the check `isDebugEnabled`, so we can skip the check here